modules/data.atmosphere/R/download.ICOS.R

```diff
@@ -0,0 +1,234 @@
+      body <- "
+      prefix cpmeta: <http://meta.icos-cp.eu/ontologies/cpmeta/>
+      prefix prov: <http://www.w3.org/ns/prov#>
+      select ?dobj ?spec ?timeStart ?timeEnd
+      where {
+      	VALUES ?spec {<data_type>}
+      	?dobj cpmeta:hasObjectSpec ?spec .
+      	VALUES ?station {<http://meta.icos-cp.eu/resources/stations/ES_sitename>}
+      			?dobj cpmeta:wasAcquiredBy/prov:wasAssociatedWith ?station .
+      ?dobj cpmeta:hasStartTime | (cpmeta:wasAcquiredBy / prov:startedAtTime) ?timeStart .
+      ?dobj cpmeta:hasEndTime | (cpmeta:wasAcquiredBy / prov:endedAtTime) ?timeEnd .
+      	FILTER NOT EXISTS {[] cpmeta:isNextVersionOf ?dobj}
+      }
+      "
+      body <- gsub("data_type", data_type, body)
+      body <- gsub("sitename", sitename, body)
+      response <- httr::POST(url, body = body)
+      response <- httr::content(response, as = "text")
+      response <- jsonlite::fromJSON(response)
+      dataset_url <- response$results$bindings$dobj$value
+      dataset_start_date <-
+        lubridate::as_datetime(
+          strptime(response$results$bindings$timeStart$value, format = "%Y-%m-%dT%H:%M:%S")
+        )
+      dataset_end_date <-
+        lubridate::as_datetime(
+          strptime(response$results$bindings$timeEnd$value, format = "%Y-%m-%dT%H:%M:%S")
+        )
+      if (is.null(dataset_url)) {
+        PEcAn.logger::logger.severe("Data is not available for the requested site")
+      }
+      if (dataset_start_date > lubridate::as_datetime(start_date)) {
+        PEcAn.logger::logger.severe(
+          paste(
+            "Data is not available for the requested start date. Please try again with",
+            dataset_start_date,
+            "as start date."
+          )
+        )
+      }
+      
+      if (dataset_end_date < lubridate::as_date(end_date)) {
+        PEcAn.logger::logger.severe(
+          paste(
+            "Data is not available for the requested end date. Please try again with",
+            dataset_end_date,
+            "as end date."
+          )
+        )
+      }
+      dataset_id <- sub(".*/", "", dataset_url)
+      
+      # construct the download URL
+      download_url <-
+        paste0('https://data.icos-cp.eu/licence_accept?ids=%5B%22',
+               dataset_id,
+               '%22%5D')
+      # Download the zip file
+      file <-
+        httr::GET(url = download_url,
+                  httr::write_disk(zip_file_name,
+                                   overwrite = TRUE),
+                  httr::progress())
+    }
+    
+    if (extract_file_flag) {
+      # extract only the hourly data file
+      zipped_csv_name <-
+        grep(
+          paste0('*', file_name),
+          utils::unzip(zip_file_name, list = TRUE)$Name,
+          ignore.case = TRUE,
+          value = TRUE
+        )
+      utils::unzip(zip_file_name,
+                   files = zipped_csv_name,
+                   junkpaths = TRUE,
+                   exdir = outfolder)
```

Comment:
I guess for drought2018, it's safe to assume this could always be the case?
```suggestion
                   exdir = outfolder)
      if (tolower(product) == "drought2018") {
        output_file_name <- zipped_csv_name
      }
```

Reply:
Yes